subsystems/sidecar/index.js

```diff
@@ -467,6 +467,13 @@ class Sidecar extends ReadyResource {
     return permits.set('trusted', Array.from(trusted))
   }
 
+  async untrust (key) {
```

Comment:
Not specific to this PR, but I believe the `permit` and this `untrust` method are susceptible to race conditions. If permit/untrust runs overlap, entries can be lost, because of the async `permits.get` and `permits.set` methods.

Note: I'm not 100% sure they can overlap in practice, but I don't see any debouncing or other race-condition handling going on here, so it seems like they can.

For `permit` this is annoying, because a user might have to `permit` again. But for `untrust` this is potentially dangerous, because a user will believe he has 'untrusted' an app, while in reality it is still trusted.

Reply:
good point, perhaps to simplify it, we can add `clearPermits` to remove all permits instead of untrust a single one?

Reply:
`await permits.clear()`

Reply:
removed this fn

Reply:
fyi, there is a task to fix the race condition

https://holepunchdev.slack.com/archives/C054FE134P6/p1728905127730489?thread_ts=1728893299.926549&cid=C054FE134P6